src/Storages/StorageMerge.cpp

```diff
@@ -405,6 +434,24 @@ Pipe StorageMerge::createSources(
             // Using concat instead.
             pipe.addTransform(std::make_shared<ConcatProcessor>(pipe.getHeader(), pipe.numOutputPorts()));
 
+        if (has_database_virtual_column)
```

Comment:
Also, could you please describe what is a `has_database_virtual_column` and why do you need one?

Reply:
Since currently, we have tables in different databases, so I think we should add a new virtual column: `_database`.